File: backend/src/models/Specialty.js
const mongoose = require('mongoose');

const specialtySchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      unique: true,
      trim: true,
    },

    description: {
      type: String,
      default: '',
      trim: true,
    },

    status: {
      type: String,
      enum: ['ACTIVE', 'INACTIVE'],
      default: 'ACTIVE',
    },

    iconUrl: {
      type: String,
      default: '',
      trim: true,
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model('Specialty', specialtySchema);

File: app/lib/booking_app/models/specialty.dart
class Specialty {
  final String id;
  final String name;
  final String description;
  final String iconUrl;
  final String status;
  const Specialty({required this.id, required this.name, required this.description, required this.iconUrl, required this.status});
  factory Specialty.fromJson(Map<String, dynamic> j) => Specialty(
    id: '${j['_id'] ?? j['id'] ?? ''}', name: '${j['name'] ?? ''}', description: '${j['description'] ?? ''}',
    iconUrl: '${j['iconUrl'] ?? ''}', status: '${j['status'] ?? 'ACTIVE'}',
  );
}
